futurex_openedx_extensions/helpers/tenants.py

```diff
@@ -314,3 +316,46 @@ def get_tenants_sites(tenant_ids: List[int]) -> List[str]:
 def get_nafath_sites() -> List:
     """Get all nafath sites"""
     return get_all_tenants_info()['special_info']['nafath_sites']
+
+
+def generate_tenant_config(name: str) -> dict:
+    """
+    Generate a tenant configuration by copying the default config and replacing placeholder PARAM.
+
+    :param name: Name to replace 'PARAM' in the default configuration
+    :type name: str
+    :return: Updated tenant configuration with placeholders replaced
+    :rtype: dict
+    """
+    try:
+        default_config = TenantConfig.objects.get(route__domain=settings.FX_DEFAULT_TENANT_SITE)
+        config_lms_dict = default_config.lms_configs
+        updated_config = json.loads(json.dumps(config_lms_dict).replace('{{platform_name}}', name))
+        return updated_config
+    except TenantConfig.DoesNotExist as exc:
+        raise FXCodedException(
+            code=FXExceptionCodes.TENANT_NOT_FOUND,
+            message='Default TenantConfig not found.',
+        ) from exc
+
+
+def create_new_tenant_config(sub_domain: str) -> TenantConfig:
+    """
+    Creates a new TenantConfig and associated Route with the provided subdomain and domain name.
+
+    :param sub_domain: The subdomain to be used for the tenant
+    :return: The created TenantConfig object
+    """
+    config_data = generate_tenant_config(sub_domain)
+
+    with transaction.atomic():
+        tenant_config = TenantConfig.objects.create(
+            external_key=sub_domain,
+            lms_configs=config_data
+        )
+        Route.objects.create(
+            domain=f'{sub_domain}.local.overhang.io',
```

Comment:
add a new config `FX_TENANTS_BASE_DOMAIN`
```suggestion
            domain=f'{sub_domain}.{settings.FX_TENANTS_BASE_DOMAIN}',
```